### ISSUE TITLE
Having trouble with DNS

### ISSUE THREAD

**Issue by harperaa** [bouvet/nord-juice-shop#68]
#### Expected Behaviour
Install and configure ctfd

#### Actual Behaviour
```
~/Dropbox/ctf/nord-juice-shop [main] $ ./manage-ctfd.sh run
Context "juicy-k8s" modified.
Creating juice-shop instance for CTFd config creation ('ctfd-config-gen')
    SUCCESS
Waiting for juice-shop instance to be ready...
error: no matching resources found
```

Also, I get error when manually trying the curl line, of create_juice_shop_instance(), without -s: 
```
~/Dropbox/ctf/nord-juice-shop [main] $ curl "https://$JUICE_FQDN/$_MU"https://$JUICE_FQDN/$_MULTI_JUICER_BASE_URL/$_MULTI_JUICER_CREATE_TEAM_URL" \
    -X POST \
    -H 'Content-Type: application/json' \
    --data-raw '{}' \
    "${_CURL_SHARED_ARGS[@]}"
curl: (6) Could not resolve host: juicier.us.eastus.cloudapp.azure.com
curl: (3) URL rejected: No host part in the URL
```
That confirms I have a DNS issue, but just don't know how to set up a host in that DNS zone, before the ingress is set up.  It seems the scripts would do that...

#### Steps to Reproduce
I completed all the steps, but I think my FQDN is not reachable and I am having trouble figuring it out.  I am new to Azure, and I have set up a DNS zone of juicier.us (it now requires two parts to the DNS zone name, where you used juice1337 before, that will not work now), the location is eastus.  So, I think my FQDN should be: juicier.us.eastus.cloudapp.azure.com.  But I cannot resolve that DNS name.  Ping, host, and dig show "Unknown host".  So, I think that is my issue.  I have the following .env.local file settings. 
```# FQDN (Fully Qualified Domain Name) at which the setup is accessible (required, used for TLS and routing)
export JUICE_FQDN="juicier.us.eastus.cloudapp.azure.com"
...
# DNS name of the NGINX controller, used as <AZURE_DNS_NAME>.<AZURE_LOCATION>.cloudapp.azure.com (required)
export AZURE_DNS_NAME="juicier.us"
# Azure region in which to deploy the services
export AZURE_LOCATION="eastus"
```
That is all, if you have any ideas how to troubleshoot the DNS issue and perhaps what I am missing there, please let me know.  I am loving the scripts and am sure it is something I did wrong with the DNS... like for example, should I have an A record in that DNS zone?  I sort of expected after the scripts, A records would be made to connect all the loadbalancers, etc... but it is not.  All I have in that DNS zone is 3 NS records and the SOA record.  

Please advise.  

Thanks in advance.

**Comment by harperaa** [comment]
Not sure if related, but I just noticed the ctfd is not running properly either, it shows it is running:
```
~/Dropbox/ctf/nord-juice-shop [main] $ kubectl get pods
NAME                                                      READY   STATUS      RESTARTS   AGE
cert-manager-7557f7b4d7-9cftv                             1/1     Running     0          53m
cert-manager-cainjector-5cfd484d6d-7g7dc                  1/1     Running     0          53m
cert-manager-webhook-765cbb8b47-x7rkn                     1/1     Running     0          53m
cleanup-job-28631700-2zvfg                                0/1     Completed   0          51m
cm-acme-http-solver-2zhh5                                 1/1     Running     0          53m
ctfd-ctfd-5c764566dd-czxtb                                1/1     Running     0          54m
...
```
But when I try to do port forwarding and hit it locally, I get this:
```
~/Dropbox/ctf/nord-juice-shop [main] $ kubectl port-forward ctfd-ctfd-5c764566dd-czxtb 8080:80
Forwarding from 127.0.0.1:8080 -> 80
Forwarding from [::1]:8080 -> 80
Handling connection for 8080
E0609 00:03:17.897335   13596 portforward.go:406] an error occurred forwarding 8080 -> 80: error forwarding port 80 to pod f3913fd455a3276f22ca238c21ddd36877159c046aac4071de0998202fe8934c, uid : failed to execute portforward in network namespace "/var/run/netns/cni-2151138e-62f7-f687-1d67-c96c8e8457c4": failed to connect to localhost:80 inside namespace "f3913fd455a3276f22ca238c21ddd36877159c046aac4071de0998202fe8934c", IPv4: dial tcp4 127.0.0.1:80: connect: connection refused IPv6 dial tcp6: address localhost: no suitable address found 
E0609 00:03:17.898920   13596 portforward.go:234] lost connection to pod
~/Dropbox/ctf/nord-juice-shop [main] $ 
```

**Comment by harperaa** [comment]
I figured out my mistake... I thought I needed a DNS Zone and then used that Zone name as AZURE_DNS_NAME... that combination gave above issues... I found this good article, for newbies on Azure, on how to set up a DNS Label, using Helm (middle of page), which made me realize, I could use a DNS Label in the .env.local... and it worked!  Posting this here, for next newbie to find.  Simply use DNS Label, no need for DNS Zone, unless you have a custom domain, I guess...  So, my URL ended up being jucier.eastus.cloudapp.azure.com...just like comment in .env file said... without the DNS Zone.

https://learn.microsoft.com/en-us/azure/aks/ingress-tls?tabs=azure-cli#add-an-a-record-to-your-dns-zone
